tools/serverpod_cli/lib/src/commands/upgrade.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:io';
+
+import 'package:serverpod_cli/src/logger/logger.dart';
+import 'package:serverpod_cli/src/runner/serverpod_command.dart';
+
+import '../generated/version.dart';
+
+class UpgradeCommand extends ServerpodCommand {
+  @override
+  final name = 'upgrade';
+
+  @override
+  final description = 'Upgrade Serverpod to the latest version.';
+
+  @override
+  void run() async {
+    dynamic error;
+    var success = await log.progress(
+        'Running `dart pub global activate serverpod_cli`...', () async {
+      var process = await Process.run(
```

Comment:
Discussion: If something goes wrong and the update process hangs the user won't be informed that here is an issue as the process will never exit.

An option is to use `Process.start` and bind stdout to debug logger and stderr to the error logger. Then, we can stream that information to the user.

Here is an example of when we have done something similar for a test: https://github.com/serverpod/serverpod/blob/6e43d86c4c8028acd932b20cdd022f9aa425de92/tests/bootstrap_project/test/project_mini_test.dart#L54-L61

Reply:
Following your recommendation the error handling is as follows:
<img width="1353" alt="image" src="https://github.com/serverpod/serverpod/assets/34163765/e4acc4b8-1de0-45fc-ac8a-6cca22c8577c">

Reply:
Note that the transforms should be connected before you `await` the `startProcess.exitCode`. We want to bind them both in all cases so there is no need to have an if statement guarding them.

Otherwise I think we are stuck in the same situation where no message is logged until the process has exited since we wait for the process to exit before binding the loggers.